packages/treasury/test/swingsetTests/governance/vat-voter.js

```diff
@@ -0,0 +1,85 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { Far } from '@agoric/marshal';
+import { q } from '@agoric/assert';
+import { sameStructure } from '@agoric/same-structure';
+import { validateQuestionFromCounter } from '@agoric/governance/src/contractGovernor';
+import { assertContractElectorate } from '@agoric/governance/src/validators';
+import { assertBallotConcernsQuestion } from '@agoric/governance/src/governParam';
+
+const build = async (log, zoe) => {
+  return Far('voter', {
+    createVoter: async (name, invitation) => {
+      const seat = E(zoe).offer(invitation);
+      const voteFacet = E(seat).getOfferResult();
+
+      return Far(`Voter ${name}`, {
+        castBallotFor: async (handle, choice) => {
+          log(`Voter ${name} cast a ballot for ${q(choice)}`);
+          return E(voteFacet).castBallotFor(handle, [choice]);
+        },
+        validate: async (
+          counterInstance,
+          governedInstance,
+          electorateInstance,
+          governorInstance,
+          issue,
+        ) => {
+          // I'd like to validate Installations, but there doesn't seem to be a
+          // way to get it from an Instance. I'd verify the Electorate,
+          // ballotCounter, and contractGovernor.
```

Comment:
What about [E(zoe).getInstallationForInstance(instance)](https://agoric.com/documentation/zoe/api/zoe.html#e-zoe-getinstallationforinstance-instance)?

Reply:
Thanks. This comment predates both the implementation of that feature, and I think the issue that tracked it.